core/src/types/operator/registry.rs

```diff
@@ -0,0 +1,226 @@
+        #[cfg(feature = "services-onedrive")]
+        registry.register_builder::<Onedrive>();
+        #[cfg(feature = "services-postgresql")]
+        registry.register_builder::<Postgresql>();
+        #[cfg(feature = "services-gdrive")]
+        registry.register_builder::<Gdrive>();
+        #[cfg(feature = "services-oss")]
+        registry.register_builder::<Oss>();
+        #[cfg(feature = "services-persy")]
+        registry.register_builder::<Persy>();
+        #[cfg(feature = "services-redis")]
+        registry.register_builder::<Redis>();
+        #[cfg(feature = "services-rocksdb")]
+        registry.register_builder::<Rocksdb>();
+        #[cfg(feature = "services-s3")]
+        registry.register_builder::<S3>();
+        #[cfg(feature = "services-seafile")]
+        registry.register_builder::<Seafile>();
+        #[cfg(feature = "services-upyun")]
+        registry.register_builder::<Upyun>();
+        #[cfg(feature = "services-yandex-disk")]
+        registry.register_builder::<YandexDisk>();
+        #[cfg(feature = "services-pcloud")]
+        registry.register_builder::<Pcloud>();
+        #[cfg(feature = "services-sftp")]
+        registry.register_builder::<Sftp>();
+        #[cfg(feature = "services-sled")]
+        registry.register_builder::<Sled>();
+        #[cfg(feature = "services-sqlite")]
+        registry.register_builder::<Sqlite>();
+        #[cfg(feature = "services-supabase")]
+        registry.register_builder::<Supabase>();
+        #[cfg(feature = "services-swift")]
+        registry.register_builder::<Swift>();
+        #[cfg(feature = "services-tikv")]
+        registry.register_builder::<Tikv>();
+        #[cfg(feature = "services-vercel-artifacts")]
+        registry.register_builder::<VercelArtifacts>();
+        #[cfg(feature = "services-vercel-blob")]
+        registry.register_builder::<VercelBlob>();
+        #[cfg(feature = "services-webdav")]
+        registry.register_builder::<Webdav>();
+        #[cfg(feature = "services-webhdfs")]
+        registry.register_builder::<Webhdfs>();
+        #[cfg(feature = "services-redb")]
+        registry.register_builder::<Redb>();
+        #[cfg(feature = "services-mongodb")]
+        registry.register_builder::<Mongodb>();
+        #[cfg(feature = "services-hdfs-native")]
+        registry.register_builder::<HdfsNative>();
+        #[cfg(feature = "services-surrealdb")]
+        registry.register_builder::<Surrealdb>();
+        #[cfg(feature = "services-lakefs")]
+        registry.register_builder::<Lakefs>();
+        #[cfg(feature = "services-nebula-graph")]
+        registry.register_builder::<NebulaGraph>();
+
+        registry
+    }
+
+    fn register_builder<B: Builder>(&mut self) {
+        self.register(
+            B::SCHEME.into_static(),
+            operator_factory_from_configurator::<B::Config>(),
+        );
+    }
+}
+
+fn operator_factory_from_configurator<C: Configurator>() -> OperatorFactory {
```

Comment:
included this private function to reduce boilerplate of operator factory generation. Any suggestion about it or is it ok to have it here?

Reply:
I prefer to remove `operator_factory_from_configurator`.